app/tr_php/01/task06.php

```diff
@@ -23,10 +28,116 @@
  *  4冊目 -> タイトル:退屈なことはPythonにやらせよう, 値段:2,200円
 */
 
-$input;
-$input_text;
-$result;
+$input = filter_input_array(INPUT_POST, [
+    'mode' => FILTER_DEFAULT,
+    'category' => FILTER_DEFAULT,
+    'min_price' => FILTER_VALIDATE_INT,
+    'max_price' => FILTER_VALIDATE_INT,
+]);
+
+if (!is_array($input)) {
+    $input = []; // Prevent null access warnings
+}
+
+
+if (empty($input['mode']) && !empty($input['category'])) {
+    $input['mode'] = 'category';
+}
+
+
+$min = isset($input['min_price']) && $input['min_price'] !== false ? $input['min_price'] : 0;
+$max = isset($input['max_price']) && $input['max_price'] !== false ? $input['max_price'] : PHP_INT_MAX;
+
+$input_text='';
+$result='';
+
+$checked = [
+    'all' => ($input['mode'] ?? '') === 'all' ? 'checked' : '',
+    'most' => ($input['mode'] ?? '') === 'most' ? 'checked' : '',
+    'category' => ($input['mode'] ?? '') === 'category' ? 'checked' : '',
+    'price' => ($input['mode'] ?? '') === 'price' ? 'checked' : '',
+];
+
+$selected_category = $input['category'] ?? '';
+$min_price_val = isset($input['min_price']) && $input['min_price'] !== false ? $input['min_price'] : '';
+$max_price_val = isset($input['max_price']) && $input['max_price'] !== false ? $input['max_price'] : '';
+
+
+if (!empty($input['mode'])) {
+    switch ($input['mode']) {
+        case 'all':
+            $input_text = '全て';
+            $list = [];
+            foreach ($category_books as $cat => $books) {
+               $list[] = "【{$cat}】";
```

Comment:
変数名を略する場合は、他の人にも伝わるようなものにしましょう。
catは伝わりづらいので、`category`をそのまま使った方が良いですね。
```suggestion
            foreach ($category_books as $category => $books) {
               $list[] = "【{$category}】";
```

Reply:
ご指摘、ありがとうございます！
今までは他の人と協力してワークを完成することがなかったため、この部分に気づきませんでした。今後、変数名を略する場合は、他の人にも伝わるようなものにすることを心掛けて進めます。